plugins/modules/snap.py

```diff
@@ -0,0 +1,338 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024- IBM, Inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+from ansible.module_utils.basic import AnsibleModule
+import shutil
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: snap_command
+author:
+- AIX Development Team (@pbfinley1911)
+short_description: Run snap command on AIX.
+description:
+- This module facilitates running the AIX snap command to gather diagnostic data.
+  Users can specify various options like collecting performance data, clearing old data,
+  or generating a complete system report.
+version_added: '1.0.0'
+requirements:
+- AIX >= 7.1 TL3
+- Python >= 3.6
+- Root user is required.
+options:
+  action:
+    description:
+    - Specifies the command you have to run.
+    - C(snap) run the snap command for specific option .
+    type: str
+    choices: [ snap ]
```

Comment:
Only one choice of action is there, It does not make sense to get it from the user for every task.